microovn/ovn/start.go

```diff
@@ -0,0 +1,45 @@
+package ovn
+
+import (
+	"fmt"
+
+	"github.com/canonical/microcluster/state"
+)
+
+// Start will update the existing OVN central and OVS switch configs.
+func Start(s *state.State) error {
+	// Skip if the database isn't ready.
+	if !s.Database.IsOpen() {
+		return nil
+	}
+
+	// Make sure the storage exists.
+	err := createPaths()
+	if err != nil {
+		return err
+	}
+
+	// Re-generate the configuration.
+	err = generateEnvironment(s)
+	if err != nil {
+		return fmt.Errorf("Failed to generate the daemon configuration: %w", err)
+	}
+
+	// Reconfigure OVS to use OVN.
```

Comment:
Is there a reason why we don't reconfigure OVN NB and SB as well here? Calling `NBCtl` and `SBCtl` here to configure listening address could help us smooth out the upgrade process going from "cli-based" to "in-db" listen configuration.

Reply:
I don't really like the idea of running commands on every start which we know should never ever change, that just ends up needlessly burning CPU and time on every startup for every user.

What we've done for situations like these in LXD is keep a list of `patches` that have been applied to a particular system, either in the database or in a text file on disk. On startup, we load up the list of patches that have been applied and compare it with the list of patches in the code, anything that's not been applied yet gets applied.

Those patches are simple Go functions and are in charge of dealing with things like relocating data or reconfiguring things, basically actions which must be performed on every system but only be performed once.

Reply:
Oh yeah, makes sense since right now the listen string is pretty much static. I was thinking ahead to when we'll use `Local_Config` database and listen strings with static IP, but we'll cross that bridge when we'll get to it. :+1:

I think that for the "upgrade problem" we can just utilize [refresh hooks](https://snapcraft.io/docs/supported-snap-hooks#heading--post-refresh) from snap.